server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -722,6 +722,12 @@ public CompletionStage<Result> addFileWithApplicantId(
               if (keysOptional.isPresent()) {
                 ImmutableList<String> keys = keysOptional.get();
 
+                if (!fileUploadQuestion.canUploadFile()) {
+                  return failedFuture(
+                      new IllegalArgumentException(
+                          "Cannot upload additional files for this question."));
```

Comment:
Done.

And yeah you understand correctly, when they hit the limit the button just disappears (and re-appears if they remove a file). I want to run this by folks to see how it feels and what - if anything - we'd like to do differently as I imagine it might be a little confusing.